Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -252,7 +281,14 @@ self_intersections_impl(const FaceRange& face_range,
       if(throw_on_SI)
         throw CGAL::internal::Throw_at_output_exception();
       else
+      {
         *out++= std::make_pair(f, f);
+        ++counter;
+        if(do_limit && counter == maximum_number)
+        {
+          return out;
+        }
```

Comment:
Edge case, but if I pass `0` the current code will limit and dump a single self-intersection (same at the other place).